src/routing/error_cause.rs

```diff
@@ -129,13 +130,26 @@ impl ErrorCause {
             Self::RateLimited(_) => ErrorClientFacing::RateLimited,
             Self::HttpGatewayError(x) => match x {
                 HttpGatewayError::AgentError(y) => {
-                    let error_string = y.to_string();
-                    if error_string.contains("no_healthy_nodes") {
-                        return ErrorClientFacing::SubnetUnavailable;
-                    } else if error_string.contains("canister_not_found") {
-                        return ErrorClientFacing::CanisterIdNotFound;
+                    match y.as_ref() {
+                        AgentError::CertifiedReject(z) | AgentError::UncertifiedReject(z) => {
+                            if z.reject_code == RejectCode::CanisterError {
```

Comment:
i suggest to also nested inner `match` pattern for consistency instead of mixing `match` and `if-else`